ApplicationLibCode/ModelVisualization/Intersections/RivExtrudedCurveIntersectionGeometryGenerator.cpp

```diff
@@ -139,21 +140,62 @@ void RivExtrudedCurveIntersectionGeometryGenerator::calculateSegementTransformPr
 //--------------------------------------------------------------------------------------------------
 ///
 //--------------------------------------------------------------------------------------------------
-void RivExtrudedCurveIntersectionGeometryGenerator::calculateFlattenedOrOffsetedPolyline()
+void RivExtrudedCurveIntersectionGeometryGenerator::calculateTransformedPolyline()
 {
-    CVF_ASSERT( m_segementTransformPrLinePoint.size() == m_polyLines.size() );
+    CVF_ASSERT( m_lineSegmentTransforms.size() == m_polylines.size() );
 
-    for ( size_t pLineIdx = 0; pLineIdx < m_polyLines.size(); ++pLineIdx )
+    for ( size_t lineIdx = 0; lineIdx < m_polylines.size(); ++lineIdx )
     {
-        m_flattenedOrOffsettedPolyLines.emplace_back();
-        const std::vector<cvf::Vec3d>& polyLine = m_polyLines[pLineIdx];
+        auto        flatPolyline = m_transformedPolyLines.emplace_back();
+        const auto& polyline     = m_polylines[lineIdx];
+
+        CVF_ASSERT( polyline.size() == m_polylines[lineIdx].size() );
```

Comment:
This assert is strange: polyLine and m_polyines[lineIdx] is the same object I think?